src/mindlessgen/prog/config.py

```diff
@@ -735,6 +737,66 @@ def __init__(self, config_file: str | Path | None = None):
         if config_file:
             self.load_from_toml(config_file)
 
+    def check_config(self, verbosity: int = 1) -> None:
+        """
+        Checks ConfigClass for any incompatibilities that are imaginable
+        """
+
+        # lower number of the available cores and the configured parallelism
+        num_cores = min(mp.cpu_count(), self.general.parallel)
+        if self.general.parallel > mp.cpu_count():
+            warnings.warn(
+                f"Number of cores requested ({self.general.parallel}) is greater "
+                + f"than the number of available cores ({mp.cpu_count()})."
+                + f"Using {num_cores} cores instead."
+            )
+
+        if num_cores > 1 and self.postprocess.debug:
+            # raise warning that debugging of postprocessing will disable parallelization
+            warnings.warn(
+                "Debug output might seem to be redundant due to the parallel processes "
+                + "with possibly similar errors in parallel mode. "
+                + "Don't be confused!"
+            )
+
+        if verbosity > 0:
+            if num_cores > 1:
+                # raise warning that parallelization will disable verbosity
+                warnings.warn(
+                    "Parallelization will disable verbosity during iterative search. "
+                    + "Set '--verbosity 0' or '-P 1' to avoid this warning, or simply ignore it."
+                )
+
+            # Check for f-block elements in forbidden elements
+            if self.generate.forbidden_elements:
+                f_block_elements = set(range(56, 70)) | set(range(88, 102))
+                if any(
+                    elem not in f_block_elements
+                    for elem in self.generate.forbidden_elements
+                ):
+                    warnings.warn(
+                        "f-block elements could be within the molecule. xTB does not treat f electrons explicitly. In this case UHF is set to 0."
+                    )
+
+            # Check for super heavy elements in forbidden elements
+            super_heavy_elements = set(range(86, 102))
+            if self.generate.element_composition and any(
+                elem in super_heavy_elements
+                for elem in self.generate.element_composition
+            ):
+                warnings.warn(
+                    "xTB does not treat super heavy elements. Approximation: atomic numbers are reduced by 32. MindlesGen terminates normally."
+                )
+
+            # Check if postprocessing is turned off
+            if not self.general.postprocess and any(
+                elem in super_heavy_elements
+                for elem in self.generate.element_composition
```

Comment:
Are you sure that the code works like this and that you don't have to iterate over the keys and values separately similar to the following code snippet:

```python
    def check_composition():
        # Align with the given element_composition:
        # CAUTION: The setting to min/max count may violate the metal count restrictions
        for elem, count_range in cfg.element_composition.items():
```

If you receive no linting/`mypy` errors and you tested the code (just execute with verbosity > 1 and appropriate settings to enforce the warning when it should appear), this can still be fine, I'm just not sure since the type of `self.generate.element_composition` is the following:
```python
self._element_composition: dict[int, tuple[int | None, int | None]] = {}
```
So it's a dictionary with the keys containing the defined atom types (as integer numbers)

Reply:
This method works i tested it and the error occurre when i enforce it, and vanishes when i change the settings properly. But i see that this could be bad practice.